lib/chainer/functions/connection/embed_id.rb

```diff
@@ -0,0 +1,50 @@
+module Chainer
+  module Functions
+    module Connection
+      class EmbedIDFunction < Chainer::Function
+        def initialize(ignore_label: nil)
+          @ignore_label = ignore_label
+        end
+
+        def self.embed_id(x, w, ignore_label: nil)
+          self.new(ignore_label: ignore_label).(x, w)
+        end
+
+        def forward_cpu(inputs)
+          (x, w) = inputs
+
+          unless @ignore_label
+            return [Chainer::Utils::Array.take(w, x.to_a, axis: 0)]
+          end
+
+          mask = Numo::Bit.cast(x.flatten.to_a.map{|i| i == @ignore_label}).reshape(*x.shape)
```

Comment:
Avoid `to_a` and `map` because it requires to convert NArray into Ruby Array as much as possible.